### DISCUSSION TITLE
WIring with the JK adapter

### DISCUSSION THREAD

**Discussion by nissag** [Sleeper85/esphome-jk-bms-can#17] [Q&A]
hi i;m building a esp32 pcb and i get the jk adapter ,; but don,t understand the drawing , from left to right ,; JK bms ; ok , jk RS485 adapter , ok,,  but what is the RS485 to 3.3 v ?   A+, B-, to what ?  i have a 5k deye mono , i bought the esp32, the transfo HI Link, the TJA 1051 T , i could build without the adapter, but what is the best option ? thanks , excuse my bad english, i;m french

**Comment by Sleeper85** [comment]
Bonjour,

L'adapteur JK RS485 n'est utile que quand l'esp32 se trouve loin du JK-BMS (+ de 10m). Le plus simple est donc sans l'adaptateur. L'adapteur ajoute une couche de transformation du signal inutile.

PS: une nouvelle version 1.17.1 sera bientôt disponible avec une nouvelle logique de charge et beaucoup de nouvelles fonctions.

**Reply by nissag** [reply]
Merci beaucoup pour cette réponse rapide, c'est parfait, bonne journée
Didier
